cmd/oras/internal/option/target.go

```diff
@@ -114,26 +116,69 @@ func parseOCILayoutReference(raw string) (path string, ref string, err error) {
 	return
 }
 
+func (opts *Target) newOCIStore() (*oci.Store, error) {
+	var err error
+	opts.Path, opts.Reference, err = parseOCILayoutReference(opts.RawReference)
+	if err != nil {
+		return nil, err
+	}
+	return oci.New(opts.Path)
+}
+
+func (opts *Target) newRepository(common Common, logger logrus.FieldLogger) (*remote.Repository, error) {
+	repo, err := opts.NewRepository(opts.RawReference, common, logger)
+	if err != nil {
+		return nil, err
+	}
+	tmp := repo.Reference
+	tmp.Reference = ""
+	opts.Path = tmp.String()
+	opts.Reference = repo.Reference.Reference
+	return repo, nil
+}
+
 // NewTarget generates a new target based on opts.
 func (opts *Target) NewTarget(common Common, logger logrus.FieldLogger) (oras.GraphTarget, error) {
 	switch opts.Type {
 	case TargetTypeOCILayout:
-		var err error
-		opts.Path, opts.Reference, err = parseOCILayoutReference(opts.RawReference)
+		return opts.newOCIStore()
+	case TargetTypeRemote:
+		return opts.newRepository(common, logger)
+	}
+	return nil, fmt.Errorf("unknown target type: %q", opts.Type)
+}
+
+type resolvableDeleter interface {
```

Comment:
Should we export this?

Reply:
I think exporting `NewBlobDeleter` and `NewManifestDeleter` are already enough.

Reply:
We can add it to oras-go BTW, it's common to resolve before deletion and should be useful to other SDK callers.